# Section 0

QPM : Assignment 1

Group Members : PICAVET Valentin Student ID : 73161

# Section 1

Use Python to generate random annual returns for 100 years for this
asset. Compare the annual mean and volatility of returns from your
random data to the true 10% mean and 20%.

In [1]:
import numpy as np

mean=0.1
vol=0.2
n_years=100

np.random.seed(123) # Setting seed for reproducibility

#Generate random annual returns for 100 years
annual_returns=np.random.normal(mean,vol,n_years)

# Calculate mean and volatility of generated annual returns
generated_mean=np.mean(annual_returns)
generated_vol=np.std(annual_returns)

print(generated_mean,generated_vol)

0.10542181469807196 0.2256480940955922


In [2]:
mean=0.1
vol=0.2

mean_change=((generated_mean-mean)/generated_mean)*100
vol_change=((generated_vol-vol)/generated_vol)*100

print(mean_change,vol_change)

5.142972271536051 11.366412908733357


The annual mean and volatility of returns from the data I generated are slightly above the true 10% mean and 20% volatility. Indeed, the mean is 5.1% higher than the true mean and the volatility is 11.4% above the true volatility.

# Section 2

Now, generate random monthly returns for 100 years for this risky
asset. Compare the annual mean and volatility of returns from your
random data to the true 10% mean and 20%

In [3]:
import numpy as np

monthly_mean = 0.10 / 12  
monthly_vol = 0.20 / np.sqrt(12) 
n_years = 100
months = n_years * 12

np.random.seed(123)  # Setting seed for reproducibility

#Generate random monthly returns for 100 years
monthly_returns = np.random.normal(monthly_mean,monthly_vol, months)

# Reshape the returns to calculate annual returns
annual_returns= np.reshape(monthly_returns, (n_years, 12)).sum(axis=1)

# Calculate mean and volatility of generated annual returns
generated_monthly_mean=np.mean(annual_returns)
generated_monthly_vol=np.std(annual_returns)

print(generated_monthly_mean,generated_monthly_vol)

0.07886108435472813 0.20358836159173124


In [4]:
mean=0.1
vol=0.2

mean_change=((generated_monthly_mean-mean)/generated_monthly_mean)*100
vol_change=((generated_monthly_vol-vol)/generated_monthly_vol)*100

print(mean_change,vol_change)

-26.805256126312045 1.7625573307216857


The annual mean from the monthly data I generated is quite below the true 10% mean and the volatility is roughly the same as the true 20% volatility. Indeed, the mean is 26.8% lower than the true mean and the volatility is  1.8% above the true volatility.

# Section 3

Finally, generate random daily returns for 100 years for this risky
asset. Compare the annual mean and volatility of returns from your
random data to the true 10% mean and 20%.

In [5]:
import numpy as np

S0= 100
daily_mean = 0.10 / 252  # I assume there are 252 trading business days in a year
daily_vol = 0.20 / np.sqrt(252) 
n_years = 100
days = n_years * 252

np.random.seed(123)  # Setting seed for reproducibility

#Generate random daily returns for 100 years
daily_returns = np.random.normal(daily_mean,daily_vol,days)

# Reshape the returns to calculate annual returns
annual_returns= np.reshape(daily_returns, (n_years, 252)).sum(axis=1)

# Calculate mean and volatility of generated annual returns
generated_daily_mean=np.mean(annual_returns)
generated_daily_vol=np.std(annual_returns)

print(generated_daily_mean,generated_daily_vol)

0.11629876037315955 0.23641355852852908


In [6]:
mean=0.1
vol=0.2

mean_change=((generated_daily_mean-mean)/generated_daily_mean)*100
vol_change=((generated_daily_vol-vol)/generated_daily_vol)*100

print(mean_change,vol_change)

14.014560706290307 15.402483155015359


The annual mean from the daily data I generated is above the true 10% mean and the volatility is also abovethe true 20% volatility. Indeed, the mean is 14.0% higher than the true mean and the volatility is 15.4% above the true volatility.

# Section 4

What do you conclude from the results of the three previous
questions?

From the previous results, I can conclude that the more random returns I generate, the more the estimated annual mean and volatility, from the returns, are far from the true mean and volatility. Indeed, the estimated annual mean and volatility estimated from the daily returns are further away from the true mean and volatility than the monthly and annual returns. 